src/infrastructure/receipt/CreateReceiptFromDTO.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2019 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Address } from '../../model/account/Address';
+import {PublicAccount} from '../../model/account/PublicAccount';
+import {MosaicId} from '../../model/mosaic/MosaicId';
+import { AddressAlias } from '../../model/namespace/AddressAlias';
+import { AliasType } from '../../model/namespace/AliasType';
+import { MosaicAlias } from '../../model/namespace/MosaicAlias';
+import { NamespaceId } from '../../model/namespace/NamespaceId';
+import { ArtifactExpiryReceipt } from '../../model/receipt/ArtifactExpiryReceipt';
+import { BalanceChangeReceipt } from '../../model/receipt/BalanceChangeReceipt';
+import { BalanceTransferReceipt } from '../../model/receipt/BalanceTransferReceipt';
+import { InflationReceipt } from '../../model/receipt/InflationReceipt';
+import { Receipt } from '../../model/receipt/Receipt';
+import { ReceiptSource } from '../../model/receipt/ReceiptSource';
+import { ReceiptType } from '../../model/receipt/ReceiptType';
+import { ResolutionEntry } from '../../model/receipt/ResolutionEntry';
+import { ResolutionStatement } from '../../model/receipt/ResolutionStatement';
+import { ResolutionType } from '../../model/receipt/ResolutionType';
+import { Statement } from '../../model/receipt/Statement';
+import { TransactionStatement } from '../../model/receipt/TransactionStatement';
+import {UInt64} from '../../model/UInt64';
+
+/**
+ * @param receiptDTO
+ * @param networkType
+ * @returns {Statement}
+ * @constructor
+ */
+export const CreateStatementFromDTO = (receiptDTO, networkType): Statement => {
+    return new Statement(
+        receiptDTO.transactionStatements.map((statement) => createTransactionStatement(statement, networkType)),
+        receiptDTO.addressResolutionStatements.map((statement) => createResolutionStatement(statement, ResolutionType.Address)),
+        receiptDTO.mosaicResolutionStatements.map((statement) => createResolutionStatement(statement, ResolutionType.Mosaic)),
+    );
+};
+
+/**
+ * @param receiptDTO
+ * @param networkType
+ * @returns {Receipt}
+ * @constructor
+ */
+export const CreateReceiptFromDTO = (receiptDTO, networkType): Receipt => {
+    switch (receiptDTO.type) {
```

Comment:
Can you link with `@see` or `@link` to the catapult-server source code where types of receipts can be found ?

Reply:
Fixed